apps/journeys-admin/src/components/JourneyView/Menu/DescriptionDialog/DescriptionDialog.tsx

```diff
@@ -30,65 +29,72 @@ export function DescriptionDialog({
   const journey = useJourney()
   const { enqueueSnackbar } = useSnackbar()
 
-  const [value, setValue] = useState(
-    journey !== undefined ? journey.description : ''
-  )
-
-  const handleSubmit = async (): Promise<void> => {
+  const handleSubmit = async (values: FormikValues): Promise<void> => {
     if (journey == null) return
 
-    const updatedJourney = { description: value }
-
     try {
       await journeyUpdate({
-        variables: { id: journey.id, input: updatedJourney },
+        variables: {
+          id: journey.id,
+          input: { description: values.description }
+        },
         optimisticResponse: {
           journeyUpdate: {
             id: journey.id,
             __typename: 'Journey',
-            ...updatedJourney
+            description: values.description
           }
         }
       })
+      onClose()
     } catch (error) {
       enqueueSnackbar('There was an error updating description', {
         variant: 'error'
       })
     }
   }
 
-  const handleClose = (): void => {
-    onClose()
+  function handleClose(resetForm: (values: FormikValues) => void): () => void {
+    return () => {
+      resetForm({ values: { description: journey?.description } })
+      onClose()
+    }
   }
 
   return (
     <>
-      <Dialog
-        open={open}
-        handleClose={handleClose}
-        dialogTitle={{ title: 'Edit Description' }}
-        dialogAction={{
-          onSubmit: handleSubmit,
-          closeLabel: 'Cancel'
-        }}
-      >
-        <form onSubmit={handleSubmit}>
-          <FormControl component="fieldset" sx={{ width: '100%' }}>
-            <FormLabel
-              component="legend"
-              aria-label="form-update-description"
-            />
-            <TextField
-              hiddenLabel
-              value={value}
-              multiline
-              variant="filled"
-              rows={3}
-              onChange={(e) => setValue(e.currentTarget.value)}
-            />
-          </FormControl>
-        </form>
-      </Dialog>
+      {journey != null && (
+        <Formik
+          initialValues={{ description: journey.description ?? '' }}
+          onSubmit={handleSubmit}
```

Comment:
```suggestion
          onSubmit={handleUpdateDescription}
```

And do the same for title so don't confuse between our submit function and the one passed by formik.